indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -20,36 +21,67 @@ import {
 } from 'indico/react/components';
 import {useNumericParam} from 'indico/react/util/routing';
 import {Translate} from 'indico/react/i18n';
+import {useIndicoAxios} from 'indico/react/hooks';
 import {EditableType} from '../../models';
 import StateIndicator from '../../editing/timeline/StateIndicator';
 
 import './EditableList.module.scss';
 
 export default function EditableList() {
-  const list = []; // Replace it with list from server
   const eventId = useNumericParam('confId');
   const {type} = useParams();
-  const [sortBy, setSortBy] = useState('id');
+  const {data, loading: isLoadingEditableList, lastData} = useIndicoAxios({
+    url: editableListURL({confId: eventId, type}),
+    camelize: true,
+    trigger: eventId,
+  });
+
+  let editableList = data || lastData;
+  if (isLoadingEditableList && !lastData) {
+    return <Loader inline="centered" active />;
+  } else if (!editableList) {
+    return null;
+  }
+
+  editableList = [...data];
+  editableList.forEach(contribution => {
+    if (_.isEmpty(contribution.editable)) {
+      contribution.status = 'not_submitted';
+      contribution.editor = null;
+    } else {
+      contribution.status = contribution.editable.state;
+      contribution.editor = contribution.editable.editor.name;
+    }
+    if (!contribution.code) {
+      contribution.code = 'N/A';
```

Comment:
Wouldn't it be better to actually move this to `renderCell` instead of modifying the object directly?

```javascript
{dataKey === 'code' && !code && Translate.string('N/A')}
```

Reply:
In `renderCell`, I am checking `sortedList[rowIndex][dataKey]` already, so I thought of adding info in the object itself.

Reply:
But it is a bit dirty. In case you e.g. access `contribution.code` later in the code, you would get `N/A` instead of null|undefined. IMO this shouldn't modify the data at all.

Reply:
Yes, please don't mutate data (unless it's something known to be regenerated each render).

Reply:
Considering this, should I remove the addition of `status` and `editor` attribute in object and add it in `renderCell` instead?

Reply:
yes, this is also mutating the object